### ISSUE TITLE
[otel] adding a Bytes SpanId to a record breaks loki->tempo integration

### ISSUE THREAD

**Issue by gquintard** [grafana/grafana#91818]
### What happened?

I'm using https://pkg.go.dev/go.opentelemetry.io/otel/log#Record.AddAttributes to add the `SpanId`/`TraceId` to my logs:
```
record.AddAttributes(apilog.Bytes("SpanId", sp.spanID[:]))
record.AddAttributes(apilog.Bytes("TraceId", sp.traceID[:]))
```

when this is sent to `loki`, it's properly saved as a byte sequence:
![image](https://github.com/user-attachments/assets/df58846b-7273-4f7a-b8a4-831e589feb7a)

but sadly, that breaks derived fields:
![image](https://github.com/user-attachments/assets/28d9620d-7b54-44c1-8943-849f285873a8)

which also makes sense, I guess, since `tempo` expects a string.

Right now I'm working around it by tweaking my code to
```
record.AddAttributes(apilog.String("SpanId", sp.spanID.String()))
record.AddAttributes(apilog.String("TraceId", sp.traceID.String()))
```

but this probably breaks non Grafana Labs tools

### What did you expect to happen?

to Tempo or Loki to magically convert/understand hex-encoded blobs?

### Did this work before?

I don't think so

### How do we reproduce it?

1. set up derived fields in Loki 
![image](https://github.com/user-attachments/assets/533054f9-719b-4edd-b15a-5e98b3674d4b)

2. send records to Loki with a byte sequence as SpanId/TraceID 
3. try to find the trace from a log

### Is the bug inside a dashboard panel?

no

### Environment (with versions)?

Grafana: `grafana/grafana-enterprise`:latest from the Docker Hub
OS: BTW I use Arch
Browser: firefox

### Grafana platform?

Docker

### Datasource(s)?

loki, tempo

**Comment by ifrost** [comment]
@joe-elliott if I understand correctly there's no way to query byte strings which is what is being stored in Tempo in this case. Do you think there's anything that can be done about it or should it be stored as a string?

**Comment by joe-elliott** [comment]
> no way to query byte strings which is what is being stored in Tempo in this case.

Yes, Tempo can query the bytes of a trace id. We just are talking about different representations of the bytes. Tempo supports what we believe is the more human readable/friendly hex string. The encoding shown above is base64 encoding.

**Comment by gquintard** [comment]
@joe-elliott, it's probably in a documentation page somewhere, could you share how to set up derived fields, as `__value.raw` will be hex-encoded, but `span:id` needs to be a byte sequence?
